### PR TITLE
PAYARA-3355 HK2 JDK-11 support

### PR DESCRIPTION
**Pull request by jGauravGupta**
Upstream PR : https://github.com/eclipse-ee4j/glassfish-hk2/pull/435

### CODE REVIEW

**Comment by jGauravGupta** [comment]
JAXB updated to latest version, as with v2.3.1 few tests are failing.

File: hk2-core/pom.xml [hunk 1]
```diff
@@ -74,6 +74,13 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <argLine>-enableassertions --add-exports java.base/jdk.internal.loader=ALL-UNNAMED</argLine>
```
Comment (Cousjava) [inline hk2-core/pom.xml]:
```suggestion
                     <argLine>-XX:+IgnoreUnrecognizedVMOptions -enableassertions --add-exports="java.base/jdk.internal.loader=ALL-UNNAMED"</argLine>
```

Comment (Cousjava) [reply hk2-core/pom.xml]: Change this to allow for compiling on JDK8

**Review by Cousjava — CHANGES_REQUESTED** [review]
Test fails in osgi-adapter-test on 
`Option org.ops4j.pax.exam.options.FrameworkStartLevelOption has not been recognized.`

**Pull request merged**